model/vcs/metrics.yaml

```diff
@@ -0,0 +1,147 @@
+groups:
+  - id: metric.vcs.change.count
+    type: metric
+    metric_name: vcs.change.count
+    brief: 'The number of changes (pull requests/merge requests/changelists) in a repository, categorized by their state (e.g. open or merged)'
+    instrument: updowncounter
+    unit: "{change}"
+    stability: experimental
+    attributes:
+      - ref: vcs.change.state
+        requirement_level: required
+      - ref: vcs.repository.url.full
+        requirement_level: required
+  - id: metric.vcs.change.duration
+    type: metric
+    metric_name: vcs.change.duration
+    brief: 'The time duration a change (pull request/merge request/changelist) has been in a given state.'
+    instrument: gauge
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.change.state
+        requirement_level: required
+  - id: metric.vcs.change.time_to_approval
+    type: metric
+    metric_name: vcs.change.time_to_approval
+    brief: 'The amount of time it took a change (pull request/merge request/changelist) to go from open to approved'
+    instrument: gauge
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+  - id: metric.vcs.repository.count
+    type: metric
+    metric_name: vcs.repository.count
+    brief: 'The number of repositories in an organization'
+    instrument: updowncounter
+    unit: "{repository}"
+    stability: experimental
+    attributes: []
+  - id: metric.vcs.ref.count
+    type: metric
+    metric_name: vcs.ref.count
+    brief: 'The number of refs of type branch or tag in a repository'
+    instrument: updowncounter
+    unit: "{ref}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+  - id: metric.vcs.change.lines_delta
+    type: metric
+    metric_name: vcs.change.lines_delta
+    brief: 'The number of lines added/removed in a change relative to the target branch of the change'
+    instrument: gauge
+    unit: "{line}"
+    stability: experimental
+    attributes:
+      - ref: vcs.repository.url.full
+        requirement_level: required
+      - ref: vcs.ref.name
+        requirement_level: required
+      - ref: vcs.ref.type
+        requirement_level: required
+      - ref: vcs.line_change.type
+        requirement_level: required
+  - id: metric.vcs.ref.lines_delta
+    type: metric
+    metric_name: vcs.ref.lines_delta
+    brief: 'The number of lines added/removed in a ref (branch) relative to the previous day'
```

Comment:
I think this needs to be relative to the originating ref. That's the only real way to compare ref's. I don't think we need a separate `lines_delta` for `change` and `ref`. A change will always be a `ref`. I think if we want to associate it with a change, then we add context (ie. attribute recommended of `vcs.change.id` to the `ref.lines_delta` instead of building a separate metric.

Reply:
While going over the brief for `vcs.change.lines_delta` I actually noticed an inaccuracy that I fixed in 46d61949fd7be630c6d7d08962157438e217ed49. This aligns better with your comment to compare relative to an originating ref.

The reason I specified `vcs.ref.lines_delta` to be relative to the previous day was to have heuristic that does not require us to process all commits of a branch.

What do you think? Should we combine `vcs.ref.lines_delta` and `vcs.change.lines_delta`?

Reply:
I think it'll always need to be compared by ref. changes are just a layer on top of refs, an ask to combine. After re-evaluating, I do think these should be combined. If it's associated with a change, then a change.id attribute can be added.

Reply:
I have updated the brief and combined `vcs.ref.lines_delta` and `vcs.change.lines_delta` in 8f8bb73943ce328f172f402b0683b45412073c2f.